lib/spack/spack/database.py

```diff
@@ -815,14 +896,38 @@ def query(self, query_spec=any, known=any, installed=True, explicit=any):
             # Abstract specs require more work -- currently we test
             # against everything.
             results = []
+
+            # Take care of managing the keyword arguments to the query
+            known = kwargs.pop('known', any)
+            installed = kwargs.pop('installed', True)
+            explicit = kwargs.pop('explicit', any)
+            start_date = kwargs.pop('start_date', datetime.datetime.min)
+            end_date = kwargs.pop('end_date', datetime.datetime.max)
+
+            # Raise an exception if we have still entries in kwargs
```

Comment:
see above

Reply:
This is to prevent that typos pass through kwargs.